crates/engine/tree/src/persistence.rs

```diff
@@ -204,12 +204,14 @@ pub enum PersistenceAction {
 pub struct PersistenceHandle {
     /// The channel used to communicate with the persistence task
     sender: Sender<PersistenceAction>,
+    /// The last persisted block number.
+    last_persisted_block_number: u64,
```

Comment:
This makes `PersistenceHandle` stateful (and the methods that modify this field should receive `&mut self`) let me know wdyt, maybe better to store it in the tree?

Reply:
Yeah I think so?